### ISSUE TITLE
Following "Authenticating to AWS API" guide and trying to create AWS resource gives `ValidationError: Request ARN is invalid`

### ISSUE THREAD

**Issue by kiich** [crossplane/provider-aws#517]
### What happened?
Trying to use EKS iam with serviceaccount by following the guide at https://github.com/crossplane/provider-aws/blob/master/AUTHENTICATION.md.

Everything seems to ok but when i create a RDS resource, i get:
```
kubectl describe rdsinstance rdspostgresql

Status:
  At Provider:
    Db Subnet Group:
    Endpoint:
    Pending Modified Values:
      Pending Cloudwatch Logs Exports:
  Conditions:
    Last Transition Time:  2021-01-29T16:44:24Z
    Message:               connect failed: ValidationError: Request ARN is invalid
                           status code: 400, request id: 357263cb-2c80-4978-b199-82a0fdb41e25
    Reason:                ReconcileError
    Status:                False
    Type:                  Synced
```

### How can we reproduce it?
Just followed the steps in the guide.

Have noticed the cat ... EOF script does add a suspicious looking `\` as part of role annotation but tried both ways but still failing.

Once everything configured, i simply did:
kubectl apply -f rds.yaml
where rds.yaml is:
```apiVersion: database.aws.crossplane.io/v1beta1
kind: RDSInstance
metadata:
  name: rdspostgresql
spec:
  forProvider:
    region: us-west-2
    dbInstanceClass: db.t2.small
    masterUsername: masteruser
    allocatedStorage: 20
    engine: postgres
    engineVersion: "9.6"
    skipFinalSnapshotBeforeDeletion: true
  writeConnectionSecretToRef:
    namespace: crossplane-system
    name: aws-rdspostgresql-conn```

### What environment did it happen in?
Crossplane version: alpha as per the guide page

**Comment by hasheddan** [comment]
Hi @kiich! Apologies for the delay here! The reason why you are seeing this error is because you are not specifying which `ProviderConfig` to use in your RDS instance. In the `AUTHENTICATION.md` guide, we create a `ProviderConfig` name `aws-provider`:

```yaml
apiVersion: aws.crossplane.io/v1beta1
kind: ProviderConfig
metadata:
  name: aws-provider
spec:
  credentials:
    source: InjectedIdentity
```

If you want to use it to provision a resource, you need to specify:
```
providerConfigRef:
  name: aws-provider
```

Reference: https://doc.crds.dev/github.com/crossplane/provider-aws/database.aws.crossplane.io/RDSInstance/v1beta1@v0.16.0#spec-providerConfigRef

Alternatively, you could name the `ProviderConfig` `default` and it will be used automatically :)

Feel free to close this issue if your problem has been addressed 👍

**Comment by kiich** [comment]
Ohh I see - thanks @hasheddan. I assumed as it was the ONLY provider, it will get used.
Let me try the `default` way and I'll let you know! thanks again.

**Comment by hgontijo** [comment]
Facing the same issue trying to deploy ECR:
```
apiVersion: ecr.aws.crossplane.io/v1alpha1
kind: Repository
metadata:
  name: tenant-acme-crossplane-repo
  namespace: acme
spec:
  forProvider:
    region: us-west-2
  providerConfigRef: 
    name: aws-provider
  writeConnectionSecretToRef:
    name: tenant-acme-crossplane-repo
    namespace: acme
```

```
# kubectl describe repository.ecr.aws.crossplane.io/tenant-acme-crossplane-repo
Name:         tenant-acme-crossplane-repo
Namespace:    
Labels:       <none>
Annotations:  crossplane.io/external-name: tenant-acme-crossplane-repo
API Version:  ecr.aws.crossplane.io/v1alpha1
Kind:         Repository
Metadata:
  Creation Timestamp:  2021-02-01T18:23:21Z
  Generation:          2
  Resource Version:    627888
  Self Link:           /apis/ecr.aws.crossplane.io/v1alpha1/repositories/tenant-acme-crossplane-repo
  UID:                 3a62fe2c-8efa-47ca-a675-48cad421b285
Spec:
  For Provider:
    Region:  us-west-2
    Tags:
      Key:    crossplane-kind
      Value:  repository.ecr.aws.crossplane.io
      Key:    crossplane-name
      Value:  tenant-acme-crossplane-repo
      Key:    crossplane-providerconfig
      Value:  aws-provider
  Provider Config Ref:
    Name:  aws-provider
  Write Connection Secret To Ref:
    Name:       tenant-acme-crossplane-repo
    Namespace:  acme
Status:
  At Provider:
    Registry Id:  
  Conditions:
    Last Transition Time:  2021-02-01T18:28:32Z
    Message:               connect failed: ValidationError: Request ARN is invalid
                           status code: 400, request id: 5cf08f97-5a1a-43cf-9d93-53bc000fd687
    Reason:                ReconcileError
    Status:                False
    Type:                  Synced
Events:
  Type     Reason                   Age    From                                      Message
  ----     ------                   ----   ----                                      -------
  Warning  CannotConnectToProvider  5m11s  managed/repository.ecr.aws.crossplane.io  ValidationError: Request ARN is invalid
           status code: 400, request id: 82732138-9cca-43c8-8b57-8c58160260f4
  Warning  CannotConnectToProvider  5m11s  managed/repository.ecr.aws.crossplane.io  ValidationError: Request ARN is invalid
           status code: 400, request id: 310ca738-df43-466c-9db7-c9e7f9e9a92c
  Warning  CannotConnectToProvider  5m11s  managed/repository.ecr.aws.crossplane.io  ValidationError: Request ARN is invalid
           status code: 400, request id: f93b20ba-2857-411d-bca3-739c3227404d
  Warning  CannotConnectToProvider  5m11s  managed/repository.ecr.aws.crossplane.io  ValidationError: Request ARN is invalid
           status code: 400, request id: 6dcb65a8-c71e-4540-84f9-7ae26b39f9ad
  Warning  CannotConnectToProvider  5m11s  managed/repository.ecr.aws.crossplane.io  ValidationError: Request ARN is invalid
           status code: 400, request id: f2704261-f96b-4c80-9992-a2361382e01e
  Warning  CannotConnectToProvider  5m11s  managed/repository.ecr.aws.crossplane.io  ValidationError: Request ARN is invalid
           status code: 400, request id: 447698b2-1a65-4f6f-a1a9-f1e30b238180
  Warning  CannotConnectToProvider  5m11s  managed/repository.ecr.aws.crossplane.io  ValidationError: Request ARN is invalid
           status code: 400, request id: 16708a85-8347-49d2-b9ef-83f06e592e4e
  Warning  CannotConnectToProvider  5m11s  managed/repository.ecr.aws.crossplane.io  ValidationError: Request ARN is invalid
           status code: 400, request id: 4570d8a7-f116-4969-8c56-3286ff254df2
  Warning  CannotConnectToProvider  5m11s  managed/repository.ecr.aws.crossplane.io  ValidationError: Request ARN is invalid
           status code: 400, request id: 97f771e9-7065-4020-ad0e-03ba4d51f975
  Warning  CannotConnectToProvider  11s (x3002 over 5m11s)  managed/repository.ecr.aws.crossplane.io  (combined from similar events): ValidationError: Request ARN is invalid
           status code: 400, request id: 7e913491-f014-49e3-9ba3-56733fb7b174
```

**Comment by hasheddan** [comment]
@kiich @hgontijo looks like an extra `\` may have slipped in there as you suggested (looking at docs [here](https://docs.aws.amazon.com/eks/latest/userguide/specify-service-account-role.html))

```
kubectl annotate serviceaccount -n <SERVICE_ACCOUNT_NAMESPACE> <SERVICE_ACCOUNT_NAME> \
eks.amazonaws.com/role-arn=arn:aws:iam::<AWS_ACCOUNT_ID>:role/<IAM_ROLE_NAME>
```
Could you try updating and let me know if that works okay?

**Comment by kiich** [comment]
Hey @hasheddan - indeed, i saw that from the guide
i.e.
```cat > provider-config.yaml <<EOF
apiVersion: pkg.crossplane.io/v1alpha1
kind: ControllerConfig
metadata:
  name: aws-config
  annotations:
    eks.amazonaws.com/role-arn: arn:aws:iam::$AWS_ACCOUNT_ID\:role/$IAM_ROLE_NAME. <--- the \ in this line
[...]
```
So i corrected that but still kept running into the error myself.

In the end, i tried several things - not super sure what was the fix but i tried:

1) Created a `default` provider config as your advice
2) Corrected the script that generates the ControllerConfig so it does not have the `\` in the annotations

After those 2 steps, the invalid ARN message went away but instead i started to receive a different error when i attempted to create any AWS resource which looked like:
```An error occurred (AccessDenied) when calling the AssumeRoleWithWebIdentity operation: Not authorized to perform sts:AssumeRoleWithWebIdentity```

It turned out the `trust.json` in the guide was missing this key line (well at leat for me anyways)
BEFORE:
```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
        "Federated": "arn:aws:iam::${AWS_ACCOUNT_ID}:oidc-provider/${OIDC_PROVIDER}"
      },
      "Action": "sts:AssumeRoleWithWebIdentity",
      "Condition": {
        "StringEquals": {
          "${OIDC_PROVIDER}:sub": "system:serviceaccount:${SERVICE_ACCOUNT_NAMESPACE}:${SERVICE_ACCOUNT_NAME}"
        }
      }
    }
  ]
}
```

NEW where i added 1 more conditions
```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
        "Federated": "arn:aws:iam::${AWS_ACCOUNT_ID}:oidc-provider/${OIDC_PROVIDER}"
      },
      "Action": "sts:AssumeRoleWithWebIdentity",
      "Condition": {
        "StringEquals": {
          "${OIDC_PROVIDER}:sub": "system:serviceaccount:${SERVICE_ACCOUNT_NAMESPACE}:${SERVICE_ACCOUNT_NAME}",
          "${OIDC_PROVIDER}:aud": "sts.amazonaws.com"
        }
      }
    }
  ]
}
```

After updating the above in IAM trust relationships, crossplane was able to create me AWS resources with no issues.

Now i am not super sure which one did the trick (maybe it was all 3??) but i'm just happy that things are working - it is a POC we are doing with CrossPlane so i'm happy to tear everything down and start again to pin point what fixed it for me once the POC is over.

**Comment by hgontijo** [comment]
@hasheddan, removing the extra backlash from `ControllerConfig` role-arn worked for me. Thanks!

**Comment by kiich** [comment]
Happy to make a PR @hasheddan :)

**Comment by hasheddan** [comment]
That would be great @kiich !

**Comment by hasheddan** [comment]
@kiich still interested in opening a PR? Happy to do it if you don't have bandwidth :)

**Comment by hasheddan** [comment]
Fixed by #524

**Comment by kiich** [comment]
@hasheddan 🤦  sorry, i missed this and missed out on the contribution opportunity! Next time for sure!
